Sources/Sentry/SentryHttpInterceptor.m

```diff
@@ -0,0 +1,80 @@
+#import "SentryHttpInterceptor.h"
+#import <Sentry/Sentry.h>
+
+@interface
+SentryHttpInterceptor () <NSURLSessionDelegate>
+
+@property (nonatomic, strong) NSURLSession *session;
+
+@property (nonatomic) NSString *spanId;
+@end
+
+@implementation SentryHttpInterceptor
+
++ (BOOL)canInitWithRequest:(NSURLRequest *)request
+{
+    return ![request.URL.host.lowercaseString containsString:@"sentry"]
+        && ([request.URL.scheme isEqualToString:@"http"] ||
+            [request.URL.scheme isEqualToString:@"https"]);
+}
+
++ (NSURLRequest *)canonicalRequestForRequest:(NSURLRequest *)request
+{
+    return request;
+}
+
+- (void)startLoading
+{
+    NSURLSessionConfiguration *conf = [NSURLSessionConfiguration defaultSessionConfiguration];
+    self.session = [NSURLSession sessionWithConfiguration:conf delegate:self delegateQueue:nil];
+    [[self.session dataTaskWithRequest:self.request] resume];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        self.spanId = [PerformanceTracker.instance addSpanWithOperation:self.request.URL.path
```

Comment:
What happens when we have multiple requests in-flight concurrently?

Reply:
Each request is handled in a separated thread, and schedule to create an span at the end of the UI thread, because request creation and completion may not be executed in the same thread, and PerformanceTracker needs to use the same thread for the same span.

Right now http requests are assign its own transaction.

Reply:
And now I realize that you may be wondering about the self.spanId. iOS instantiate one Interceptor per request, no concurrency problem here.

Reply:
does it mean that HTTP spans have their own transaction and they are not children of the current transaction (like a UI transaction)? how would you know that a HTTP request is slowing down your rendering process/user experience? can we make it possible as Android/JS do somehow?

Reply:
it likely has to be changed because the interceptor is called in another thread and in another moment.
if we create a span out of it at the moment its called, it might be a different transaction bound to the scope at that moment.
maybe swizzling would be a better approach but it'd require swizzling per HTTP type (URL session, etc...)